File: Leetcode/476. Number Complement/number_complement.py
class Solution:
    def findComplement(self, num: int) -> int:
        # XOR can  give us complement but issues are leading zeros so we need to xor untill the number's bit
        total_bits =0
        n=num
        while n!=0:
            n>>=1
            total_bits+=1
        mask = (1<<total_bits) -1
        return mask ^ num

sol = Solution()
print(sol.findComplement(0))

File: Leetcode/476. Number Complement/number_complement.rs
struct Solution{}

impl Solution {
    pub fn find_complement(num: i32) -> i32 {
        let mut total_bits = 0;
        let mut n = num;
        while n!=0{
            n>>=1;
            total_bits+=1
        }
        let mask = (1<<total_bits)-1;
        return num ^ mask
    }
}
fn main() {
    let res = Solution::find_complement(5);
    println!("{}",res);
}
